app/assets/javascripts/react_components/admin_dashboard/admin_recommendations.js.jsx

```diff
@@ -316,22 +472,248 @@ var CreateRecommendationPage = React.createClass({
   }
 });
 
-var RecommendationBookSearch = React.createClass({
-  _handleOnSubmit: function (e) {
-    e.preventDefault();
+var EditRecommendationPage = React.createClass({
+  getInitialState: function () {
+    var rec = this.props.recommendation;
+    var id = rec.id;
+    $.ajax({
+      url: "/admin/recommendations/" + id + "/display_proj_tags",
+      dataType: 'json',
+      success: function (data) {
+        this.setState({projectTags: data});
+      }.bind(this),
+      error: function (xhr, status, err) {
+        console.error(this.props.url, status, err.toString());
+      }.bind(this)
+    });
+
+    return {
+      recommendationType: rec.recommendation_type,
+      bookTags: [],
+      projectTags: [],
+      selectedBooks: []
+    };
+  },
+  componentDidMount: function () {
+    $('#recommendation-type-toggle').bootstrapSwitch();
+    $('#recommendation-type-toggle').on('switchChange.bootstrapSwitch', this._setRecommendationType);
+    $('.bootstrap-switch').css("float", "right");
+
+    var rec = this.props.recommendation;
+    $('.name-field').val(rec.name);
+
+    var self = this;
+    $.ajax({
+      url: "/admin/recommendations/" + rec.id + "/display_proj_tags",
+      dataType: 'json',
+      success: function (data) {
+        var projTags = [];
+        data.countries.map (function (country) {
+          var tag = _.findWhere(gon.project_tags, {id: country.id, tagType:"countries"});
+          projTags.push(tag);
+        });
+        data.languages.map (function (lang) {
+          var tag = _.findWhere(gon.project_tags, {id: lang.id, tagType:"language"});
+          projTags.push(tag);
+        });
+        self.setState({projectTags: projTags});
+        projTags.map (function (tag) {
+          $('.project-tagbar-input').tagsinput('add', tag);
+        });
+      }.bind(this),
+      error: function (xhr, status, err) {
+        console.error(this.props.url, status, err.toString());
+      }.bind(this)
+    });
+
+    if (rec.recommendation_type == RecommendationTypes.AUTO) {
+      $.ajax({
+        url: "/admin/recommendations/" + rec.id + "/display_book_tags",
+        dataType: 'json',
+        success: function (data) {
+          var bookTags = [];
```

Comment:
it seems like these success functions are very similar. Can we maybe make them a named functioned use them twice